src/main/java/com/etsy/statsd/profiler/worker/ProfilerWorkerThread.java

```diff
@@ -9,13 +13,25 @@
  */
 public class ProfilerWorkerThread implements Runnable {
     private Profiler profiler;
+    private LinkedList<String> errors;
 
-    public ProfilerWorkerThread(Profiler profiler) {
+    public ProfilerWorkerThread(Profiler profiler, LinkedList<String> errors) {
         this.profiler = profiler;
+        this.errors = errors;
     }
 
     @Override
     public void run() {
-        profiler.profile();
+        try {
+            profiler.profile();
+        } catch(Exception e) {
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            e.printStackTrace(pw);
+            errors.add(String.format("Received an error running profiler: %s, error: %s", profiler.getClass().getName(), sw.toString()));
+            if ( errors.size() > 10) {
+                errors.pollLast();
```

Comment:
I just noticed this - `add` will add the to the end of the list and `pollLast` will remove the last element of the list.  I think we want `pollFirst` instead to keep the most recent 10 errors.